tests/components/sia/test_config_flow.py

```diff
@@ -0,0 +1,170 @@
+    assert result3["title"] == "SIA Alarm on port 7777"
+    assert result3["data"] == {
+        "port": 7777,
+        "accounts": [
+            {
+                "account": "ABCDEF",
+                "encryption_key": "AAAAAAAAAAAAAAAA",
+                "ping_interval": 10,
+                "zones": 1,
+            },
+            {
+                "account": "ACC2",
+                "encryption_key": "AAAAAAAAAAAAAAAA",
+                "ping_interval": 2,
+                "zones": 2,
+            },
+        ],
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_abort_form(hass):
+    """Test we handle invalid key."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    with patch(
+        "homeassistant.config_entries.ConfigFlow._abort_if_unique_id_configured",
+        side_effect=AbortFlow(reason=ABORT_ALREADY_CONFIGURED),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], BASIC_CONFIG
+        )
+    assert result2["type"] == "abort"
+    assert result2["reason"] == ABORT_ALREADY_CONFIGURED
+
+
+@pytest.mark.parametrize(
+    "additional, field, value, error",
+    [
+        (False, "encryption_key", "AAAAAAAAAAAAAZZZ", "invalid_key_format"),
+        (True, "encryption_key", "AAAAAAAAAAAAAZZZ", "invalid_key_format"),
+        (False, "encryption_key", "AAAAAAAAAAAAA", "invalid_key_length"),
+        (True, "encryption_key", "AAAAAAAAAAAAA", "invalid_key_length"),
+        (False, "account", "ZZZ", "invalid_account_format"),
+        (True, "account", "ZZZ", "invalid_account_format"),
+        (False, "account", "A", "invalid_account_length"),
+        (True, "account", "A", "invalid_account_length"),
+        (False, "ping_interval", 1500, "invalid_ping"),
+        (True, "ping_interval", 1500, "invalid_ping"),
+        (False, "zones", 0, "invalid_zones"),
+        (True, "zones", 0, "invalid_zones"),
+    ],
+)
+async def test_form_errors(hass, additional, field, value, error):
+    """Test we handle the different invalid inputs, both in the main and in the additional flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    if not additional:
+        config = BASIC_CONFIG.copy()
+        config[field] = value
+    else:
+        add_config = BASIC_CONFIG.copy()
+        add_config["additional_account"] = True
+        result_add = await hass.config_entries.flow.async_configure(
+            result["flow_id"], add_config
+        )
+        assert result_add["type"] == "form"
+        assert result_add["errors"] == {}
+
+        config = ADDITIONAL_ACCOUNT.copy()
+        config[field] = value
+
+    with patch("homeassistant.components.sia.async_setup", return_value=True), patch(
```

Comment:
I don't think you need to patch these if the config flow doesn't complete

Reply:
Correct, good one.